### ISSUE TITLE
Intrepid2: no longer compiling in develop

### ISSUE THREAD

**Issue by csiefer2** [trilinos/Trilinos#9531]
Stuff like this:
```
/Trilinos/sandbox/Trilinos/packages/intrepid2/src/Shared/Intrepid2_Data.hpp:229:71: error: expected ‘;’ at end of member declaration
     ScalarView<DataScalar,DeviceType> zeroView_ = ZeroView<DataScalar,DeviceType>::zeroView(); // one-entry (zero); used to allow getEntry() to return 0 for off-diagonal entries in BLOCK_PL
                                                                       ^
/Trilinosnos/packages/intrepid2/src/Shared/Intrepid2_Data.hpp:229:71: error: declaration of ‘Intrepid2::ScalarView<DataScalar, DeviceType> Intrepid2::Data<DataScalar, DeviceType>::DeviceType’
/Trilinos/packages/intrepid2/src/Shared/Intrepid2_Data.hpp:198:29: error:  shadows template parm ‘class DeviceType’
   template<class DataScalar,typename DeviceType>
                             ^
/Trilinos/sandbox/Trilinos/packages/intrepid2/src/Shared/Intrepid2_Data.hpp:229:81: error: expected unqualified-id before ‘>’ token
     ScalarView<DataScalar,DeviceType> zeroView_ = ZeroView<DataScalar,DeviceType>::zeroView(); // one-entry (zero); used to allow getEntry() to return 0 for off-diagonal entries in BLOCK_PL
                                                                                 ^
/Trilinos/packages/intrepid2/src/Shared/Intrepid2_Data.hpp:229:60: error: wrong number of template arguments (1, should be 2)
     ScalarView<DataScalar,DeviceType> zeroView_ = ZeroView<DataScalar,DeviceType>::zeroView(); // one-entry (zero); used to allow getEntry() to return 0 for off-diagonal entries in BLOCK_PL
 ```

SerialNode builds.  Gcc 5.3.0 or Clang, take your pick.

**Comment by csiefer2** [comment]
@trilinos/intrepid2 @CamelliaDPG

**Comment by william76** [comment]
Looks like the Intrepid2 PR in question may be #9524 but #9525 also got merged yesterday after it. That said, #9525 _started_ before #9524 was merged so they kind of passed by each other in flight. 

Could there be some additive bug happening in which both changes were ok with the previous `develop` branch but after merging they are incompatible?  #9525 was from @trilinos/shylu and @ndellingwood reviewed it.

FYI @prwolfe @trilinos/framework

**Comment by ndellingwood** [comment]
#9525 only touched shylu-basker code and amesos2 internal solver code related to shylu-basker, nothing related or used by intrepid2

**Comment by prwolfe** [comment]
I don't see this on the dev->master board. The main serial testing is using gcc 7.2 and that's clean. The clang builds are also clean and the failures under gcc 8.3 and intel 19 are OOM kills looks like.

@csiefer2 - did you really mean gcc 5? Do you have a version of clang or a cmake fragment that someone could use to re-create this?

**Comment by CamelliaDPG** [comment]
@csiefer2, I have the same question as @prwolfe.  I'm now looking to reproduce the issue.  Thanks!

**Comment by CamelliaDPG** [comment]
(Ideally, I'd like to reproduce it with clang, since that's what I have locally.)

**Comment by csiefer2** [comment]
This will should break:

```
cmake \
-D CMAKE_BUILD_TYPE:STRING="Release" \
-D Trilinos_ENABLE_Amesos2=ON \
-D Amesos2_ENABLE_KLU2=ON \
-D Trilinos_ENABLE_AztecOO=ON \
-D Trilinos_ENABLE_Belos=ON \
-D Trilinos_ENABLE_Epetra=ON \
-D Trilinos_ENABLE_EpetraExt=ON \
-D Trilinos_ENABLE_Galeri=ON \
-D Trilinos_ENABLE_Ifpack2=ON \
-D Trilinos_ENABLE_Intrepid=ON \
-D Trilinos_ENABLE_Intrepid2=ON \
-D Trilinos_ENABLE_Kokkos=ON \
-D Trilinos_ENABLE_LOCA=OFF \
-D Trilinos_ENABLE_ML=ON \
-D Trilinos_ENABLE_MueLu=ON \
 -D MueLu_ENABLE_Boost_for_real=ON \
 -D MueLu_ENABLE_Experimental=OFF \
 -D MueLu_ENABLE_TESTS=ON \
 -D MueLu_ENABLE_EXAMPLES=ON \
-D Trilinos_ENABLE_NOX=OFF \
-D Trilinos_ENABLE_Pamgen=ON \
-D Trilinos_ENABLE_Phalanx=OFF \
-D Trilinos_ENABLE_Piro=OFF \
-D Trilinos_ENABLE_Rythmos=OFF \
-D Trilinos_ENABLE_SEACAS=OFF \
-D Trilinos_ENABLE_SEACASExodus=OFF \
-D Trilinos_ENABLE_STK=OFF \
-D Trilinos_ENABLE_STKIO=OFF \
-D Trilinos_ENABLE_STKMesh=OFF \
-D Trilinos_ENABLE_STKClassic=OFF \
-D Trilinos_ENABLE_Stratimikos=ON \
-D Trilinos_ENABLE_Stokhos=OFF \
-D Trilinos_ENABLE_Teuchos=ON \
-D Trilinos_ENABLE_ThreadPool=OFF \
-D Trilinos_ENABLE_Teuchos=ON \
-D Trilinos_ENABLE_Tpetra=ON \
 -D Tpetra_INST_INT_INT=ON \
 -D Tpetra_INST_INT_LONG_LONG=OFF \
-D Trilinos_ENABLE_TrilinosCouplings=OFF \
-D Trilinos_ENABLE_TriKota=OFF \
-D Trilinos_ENABLE_Xpetra=ON \
 -D Xpetra_ENABLE_Experimental=ON \
../Trilinos

**Comment by csiefer2** [comment]
The code seems to be breaking on
```
#include "Intrepid2_Basis.hpp"
```

MueLu does not include any Intrepid2 headers before that point.

**Comment by CamelliaDPG** [comment]
@csiefer2, I'm still having trouble reproducing.  Using Apple clang 11.0.3, and the following configuration script, I do a clean build without errors, and without any warnings from Intrepid2.

Here's my configure:
```
cmake \
-G "Ninja" \
-D CMAKE_BUILD_TYPE:STRING="Release" \
-D Trilinos_ENABLE_Amesos2=ON \
-D Amesos2_ENABLE_KLU2=ON \
-D Trilinos_ENABLE_AztecOO=ON \
-D Trilinos_ENABLE_Belos=ON \
-D Trilinos_ENABLE_Epetra=ON \
-D Trilinos_ENABLE_EpetraExt=ON \
-D Trilinos_ENABLE_Galeri=ON \
-D Trilinos_ENABLE_Ifpack2=ON \
-D Trilinos_ENABLE_Intrepid=ON \
-D Trilinos_ENABLE_Intrepid2=ON \
-D Trilinos_ENABLE_Kokkos=ON \
-D Trilinos_ENABLE_LOCA=OFF \
-D Trilinos_ENABLE_ML=ON \
-D Trilinos_ENABLE_MueLu=ON \
-D MueLu_ENABLE_Boost_for_real=ON \
-D MueLu_ENABLE_Experimental=OFF \
-D MueLu_ENABLE_TESTS=ON \
-D MueLu_ENABLE_EXAMPLES=ON \
-D Trilinos_ENABLE_NOX=OFF \
-D Trilinos_ENABLE_Pamgen=ON \
-D Trilinos_ENABLE_Phalanx=OFF \
-D Trilinos_ENABLE_Piro=OFF \
-D Trilinos_ENABLE_Rythmos=OFF \
-D Trilinos_ENABLE_SEACAS=OFF \
-D Trilinos_ENABLE_SEACASExodus=OFF \
-D Trilinos_ENABLE_STK=OFF \
-D Trilinos_ENABLE_STKIO=OFF \
-D Trilinos_ENABLE_STKMesh=OFF \
-D Trilinos_ENABLE_STKClassic=OFF \
-D Trilinos_ENABLE_Stratimikos=ON \
-D Trilinos_ENABLE_Stokhos=OFF \
-D Trilinos_ENABLE_Teuchos=ON \
-D Trilinos_ENABLE_ThreadPool=OFF \
-D Trilinos_ENABLE_Teuchos=ON \
-D Trilinos_ENABLE_Tpetra=ON \
-D Tpetra_INST_INT_INT=ON \
-D Tpetra_INST_INT_LONG_LONG=OFF \
-D Trilinos_ENABLE_TrilinosCouplings=OFF \
-D Trilinos_ENABLE_TriKota=OFF \
-D Trilinos_ENABLE_Xpetra=ON \
-D Xpetra_ENABLE_Experimental=ON \
-DCMAKE_INSTALL_PREFIX:PATH="/Users/nvrober/trilinos-builds/reproduce-9531" \
$EXTRA_ARGS \
../..
```

Could you give some more details about the particular platform on which you're seeing the failure?

**Comment by csiefer2** [comment]
RHEL7, gcc.  Lemme try your clang config and see if it works or not.

**Comment by CamelliaDPG** [comment]
@csiefer2, I tried on cee just now, with gcc 7.2.0 (the oldest I could find among the modules).  With the same config as above, it built fine.

**Comment by cgcgcg** [comment]
Not sure it helps, but some of our nightlies show the issue. E.g.
https://testing.sandia.gov/cdash/build/9723634

**Comment by csiefer2** [comment]
Yup.  Fails to compile with GCC 5.3.0 (but using the above cmake script otherwise).

**Comment by CamelliaDPG** [comment]
Is there a module I can load on cee to get 5.3.0?  I didn't immediately see one, but I might have missed it.

**Comment by jhux2** [comment]
> Is there a module I can load on cee to get 5.3.0? I didn't immediately see one, but I might have missed it.

```
module load sems-env
module load sems-gcc/5.3.0
```

**Comment by CamelliaDPG** [comment]
Thanks, @jhux2!

**Comment by CamelliaDPG** [comment]
Should be fixed in #9565.
